classes/Model/Popup.php

```diff
@@ -826,6 +826,12 @@ public function is_loadable() {
 			// Published/private
 		}
 
+		// If popup is not active, this popup is not loadable.
+		$active = $this->get_meta( 'active' );
+		if ( false === $active || '0' === $active ) {
```

Comment:
False may not be a good value to use here, nor 0 because both are converted to a falsey value. Unfortunately that is also the same thing that would be returned if the key wasn't set at all.

So by default this way all popup's would be disabled unless we also add the meta to enable them as well.

By default if no key exists it should not do anything.

I think changing the field to represent disabling the popup is better than having it represent enabling a popup.

This way if the setting is missing nothing happens and the popup continues to work.

We could just use `'disabled'` string when the "Disable popup" option is checked.

Then update to 

```
$disabled = $this->get_meta( 'disabled ' );
if ( 'disabled' === $disabled  ) {
```

Reply:
@danieliser This is incorrect. Our `get_meta` method uses `get_post_meta`. This only returns false if the post ID is invalid, which is why we check for false. If the post ID is valid and the meta key is missing, it returns an empty string. Since we have `===`, we only bail if the post ID is not valid or if the key exists and it is set to '0'. By default, all popups are therefore active, which I have tested.

Reply:
If that is the cae we still need to adjust it based on our conversation on labeling.

Disable Popup [( 0 )    ]
Disable Popup [    ( 1 )]

So you would check for a 1 value for disabled to bail.

Reply:
@danieliser You mean go with "Enable Popup", based on our conversation, right?

So, have `Popup Enabled [(0) ]` and `Popup Enabled [ (1)]` with it defaulting to on.

Reply:
Yes, the last comment I made was via slack about this, and from the screenshots of it on the All Popups screen, I do believe you were correct to go with Popup enabled with a Green indicator. The inverse would either be confusing or ugly.

So ignore my last comment here and stick with the concept that a popup is enabled when green.

This block of code should likely be made a new method in the PUM_Model_Popup as it is reused several times and always from the model anyways.

```
		$enabled = $popup->get_meta( 'enabled' );
		if ( '' === $enabled ) {
			$enabled = 1;
		} else {
			$enabled = intval( $enabled );
		}
```

I might also modify it (checking if we are on an admin screen) and explicitly setting the meta for future optimization when its detected to be not set.

When you call get_post_meta on a key that doesn't exist, it always calls directly to the DB to make sure its not there (and missing from cache).

To prevent those extra queries, we need to fill that value when its missing. Best way is passively as we go. This way it only does an extra write query once per popup and then prevents future select queries per popup.

So I'd update the above to something like below.

```
		$enabled = $popup->get_meta( 'enabled' );
		if ( '' === $enabled ) {
			$enabled = 1;
                        $popup->update_meta( 'enabled', $enabled );
		} else {
			$enabled = intval( $enabled );
		}
```

Otherwise UI looks great.

Reply:
@danieliser Yeah, I think moving that to its own method is a good call. I'll do that real quick.

Ah! You're right, it would be extra queries. Definitely makes sense to add the update_meta call there. Will do.